apps/web/src/components/happening-info-box.tsx

```diff
@@ -0,0 +1,189 @@
+import Image from "next/image";
+import Link from "next/link";
+import { notFound } from "next/navigation";
+import { ExternalLinkIcon } from "@radix-ui/react-icons";
+import { isAfter, isBefore } from "date-fns";
+
+import { Happening } from "@echo-webkom/db/schemas";
+
+import { Heading } from "@/components/ui/heading";
+import { maxCapacityBySlug } from "@/lib/queries/happening";
+import { fetchBedpresBySlug, type Bedpres } from "@/sanity/bedpres";
+import { fetchEventBySlug, type Event } from "@/sanity/event";
+import { cn } from "@/utils/cn";
+import { urlFor } from "@/utils/image-builder";
+import { capitalize } from "@/utils/string";
+
+export async function HappeningInfoBox({ happening }: { happening: Happening }) {
+  return (
+    <div>
+      {happening.type === "event" && <EventInfoBox event={await getEventBySlug(happening.slug)} />}
+      {happening.type === "bedpres" && (
+        <BedpresInfoBox bedpres={await getBedpresBySlug(happening.slug)} />
+      )}
+    </div>
+  );
+}
+
+async function getBedpresBySlug(slug: string) {
+  const bedpres = await fetchBedpresBySlug(slug);
+
+  if (!bedpres) {
+    return notFound();
+  }
+
+  return bedpres;
+}
+
+async function getEventBySlug(slug: string) {
+  const happening = await fetchEventBySlug(slug);
+
+  if (!happening) {
+    return notFound();
+  }
+
+  return happening;
+}
+
+async function EventInfoBox({ event }: { event: Event }) {
+  const isRegistrationOpen =
+    event?.registrationStart &&
+    event?.registrationEnd &&
+    isAfter(new Date(), new Date(event.registrationStart)) &&
+    isBefore(new Date(), new Date(event.registrationEnd));
+
+  return (
+    <div
+      className={cn(
+        "flex",
+        "h-full",
+        "items-center",
+        "rounded-xl",
+        "border",
+        "gap-5",
+        "p-5",
+        "bg-card",
+        "overflow-x-auto",
+        "sm:rounded-lg",
+      )}
+    >
+      <div className="overflow-x-hidden">
+        <Heading className="line-clamp-1 text-lg font-semibold md:text-2xl">
+          {capitalize(event.title)}
+        </Heading>
+        <p className="text-sm md:text-base">Arrangement</p>
+
+        <div className="py-4">
+          <div>Sted: {event.location?.name ? event.location.name : "Ikke bestemt"}</div>
+          <div>
+            Kapasitet:{" "}
+            {(await maxCapacityBySlug(event.slug)) === 0
+              ? "uendelig"
+              : await maxCapacityBySlug(event.slug)}
+          </div>
+          <div>Publisert: {new Date(event._createdAt).toLocaleDateString()}</div>
+          <div>Sist oppdatert: {new Date(event._updatedAt).toLocaleDateString()}</div>
```

Comment:
```suggestion
          <div>Publisert: {norwegianDateString(event._createdAt)}</div>
          <div>Sist oppdatert: {norwegianDateString(event._updatedAt)}</div>
```
Her ville jeg ha brukt norwegianDateString() fra `date.ts`, eller funksjon for å formatere datoen.